### ISSUE TITLE
Exception when handling Message A10A

### ISSUE THREAD

**Issue by buracc** [zefiers/Silkroad.Net#1]
An exception occurs after sending the passcode message (0x6117):

```
SRO_TR_Official_GameGami
[Efes (341)] - Status: CROWDED
[Troya (342)] - Status: CROWDED
[Aspendos (343)] - Status: FULL
[Myra (344)] - Status: CROWDED
[Assos (345)] - Status: CROWDED
[Zeugma (346)] - Status: CROWDED
[Pergamon (347)] - Status: 2
[Misis (356)] - Status: FULL
[Patara (357)] - Status: CROWDED
Sending credentials

[2116] [NuN] [Framework] [0116] [0001 bytes]
0000000000   01                                                ................

Passcode request
Entering passcode

[A117] [Ack] [Framework] [0117] [0002 bytes]
0000000000   04 01                                             ................

Passcode entered

Unhandled exception. System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'Length')
   at Silkroad.Security.Blowfish.Decrypt(Span1 input) in C:\Users\burak\RiderProjects\Silkroad.Net\Silkroad\Security\Blowfish.cs:line 272
   at Silkroad.Network.Messaging.Protocol.MessageProtocol.Decode(MessageSize size, Span1 buffer) in C:\Users\burak\RiderProjects\Silkroad.Net\Silkroad\Network\Messaging\Protocol\MessageProtocol.cs:line 60
   at Silkroad.Network.Session.ReceiveAsync() in C:\Users\burak\RiderProjects\Silkroad.Net\Silkroad\Network\Session.cs:line 252
   at Silkroad.Network.Session.RunAsync() in C:\Users\burak\RiderProjects\Silkroad.Net\Silkroad\Network\Session.cs:line 356
   at SimpleClientless.Program.Main(String[] args) in C:\Users\burak\RiderProjects\Silkroad.Net\SimpleClientless\Program.cs:line 36
   at SimpleClientless.Program.<Main>(String[] args)
```

You can see the timeline of the packets above, it goes as follows:
Send credentials (0x610A) -> Request passcode (0x2116) -> Submit passcode (0x6117) -> Check passcode (0xA117) -> Agent auth (0xA10A) [Exception thrown]

See implementation here: https://github.com/buracc/Silkroad.Net/blob/master/SimpleClientless/Program.cs#L169

**Comment by halimsamy** [comment]
Adding a breakpoint in Session.ReceiveAsync() when receiving 0xA10A and inspecting the received bytes should help a lot.

**Comment by buracc** [comment]
Managed to get it working by not slicing the buffer when trying to decrypt:


```
/// <summary>
        ///     Decodes a raw buffer into a ready to use <see cref="Message" />.
        /// </summary>
        /// <param name="size">The 2 bytes masked message size from the raw buffer.</param>
        /// <param name="buffer">The remaining message raw buffer.</param>
        /// <returns>The decoded ready to use <see cref="Message" />.</returns>
        /// <exception cref="InvalidOperationException">The message was interrupted or injected (checksum failed).</exception>
        internal Message Decode(MessageSize size, Span<byte> buffer) {
            if (size.Encrypted && this.Option.HasFlag(MessageProtocolOption.Encryption)) {
                this.Blowfish.Decrypt(buffer);
            }

            var msg = new Message(size, buffer);
            this.Validate(msg);

            return msg;
        }
```

Located in this class: https://github.com/buracc/Silkroad.Net/blob/master/Silkroad/Network/Messaging/Protocol/MessageProtocol.cs#L60

However I'm not exactly sure what I'm doing when it comes to these things and it might break other stuff. So this might not be a viable solution.

**Comment by halimsamy** [comment]
In fact, It's true that it shouldn't be sliced, since the 'buffer' param contains the encrypted data, so no need to slice, the only 2 bytes that are not encrypted was already extracted into the 'size' param.

This is actually the bug fix, I'm waiting for you to create a pull request if you want, if you don't want, just inform me to push a commit and close this issue.

Well done!

**Comment by buracc** [comment]
Fixed in [e30af49](https://github.com/zefiers/Silkroad.Net/commit/e30af4944b8d8767b1f72c577f7fd08932442e72)
